src/components/calcite-dropdown/calcite-dropdown.tsx

```diff
@@ -223,9 +223,13 @@ export class CalciteDropdown {
 
   @Listen("calciteDropdownOpen", { target: "window" })
   closeCalciteDropdownOnOpenEvent(e: Event): void {
-    if (e.target !== this.el) {
-      this.active = false;
+    const composedPath = e.composedPath();
+
+    if (composedPath.includes(this.el)) {
```

Comment:
Nitpick: `e.composedPath().includes(this.el)` since the path is only referenced once.